pkg/generate/generate.go

```diff
@@ -332,11 +325,31 @@ func manageClone(log logr.Logger, kind, namespace, name string, clone map[string
 		// attempting to clone it self, this will fail -> short-ciruit it
 		return nil, Skip, nil
 	}
+
 	// check if the resource as reference in clone exists?
 	obj, err := client.GetResource(kind, newRNs, newRName)
 	if err != nil {
 		return nil, Skip, fmt.Errorf("reference clone resource %s/%s/%s not found. %v", kind, newRNs, newRName, err)
 	}
+
+	// check if resource to be generated exists
+	newResource, err := client.GetResource(kind, namespace, name)
+	if err == nil {
+		obj.SetUID(newResource.GetUID())
+		obj.SetSelfLink(newResource.GetSelfLink())
+		obj.SetCreationTimestamp(newResource.GetCreationTimestamp())
+		obj.SetManagedFields(newResource.GetManagedFields())
+		obj.SetResourceVersion(newResource.GetResourceVersion())
+		return obj.UnstructuredContent(), Update, nil
```

Comment:
@JimBugwadia @realshuting Before returning obj. Adding a check for a compare object can  help us to reduce update in config-map